teamcity-server/src/main/java/com/cyberark/server/ConjurBuildStartContextProcessor.java

```diff
@@ -47,39 +56,37 @@ private BuildProblemData createBuildProblem(SBuild build, String message) {
     public void updateParameters(BuildStartContext context) {
         SRunningBuild build = context.getBuild();
 
-        SBuildType buildType = build.getBuildType();
-        if (buildType == null) {
-            // It is possible of build type to be null, if this is the case let's return and not retrieve conjur secrets
-            return;
-        }
-
-        SProject project = buildType.getProject();
-        SProjectFeatureDescriptor connectionFeature = null;
-
+        Map<String, String> conjurConnParams = null;
         try {
-            connectionFeature = getConnectionType(project, ConjurSettings.getFeatureType());
-        } catch (MultipleConnectionsReturnedException e) {
+            conjurConnParams = getConjurConnectionParams(build, ConjurSettings.getFeatureType(), false);
+        } catch (MultipleConnectionsReturnedException | MissingMandatoryParameterException e) {
             BuildProblemData buildProblem = createBuildProblem(build, String.format("ERROR: %s", e.getMessage()));
             build.addBuildProblem(buildProblem);
         }
-        
-        if (connectionFeature == null) {
-            // If connection feature cannot be found (no connection has been configured on this project)
-            // then return and do not perform conjur secret retrieval actions
-            return;
-        }
 
-        ConjurConnectionParameters conjurConnParams = new ConjurConnectionParameters(connectionFeature.getParameters(), false);
+        if (conjurConnParams != null) {
+            for (Map.Entry<String, String> kv : conjurConnParams.entrySet()) {
+                context.addSharedParameter(kv.getKey(), kv.getValue());
+            }
+        }
+    }
 
+    // the purpose of this method is to mark a given param as password-type
+    @Override
+    public Collection<Parameter> getPasswordParameters(SBuild build) {
```

Comment:
How is this not adding every single param as a password type? I don't see any place where you're calling out the API key specifically, but also I'm not super familiar with modern Java. :)

Reply:
It's a decision to make - which of those parameters are sensitive enough to be marked as password-type. Before this PR, none of them was marked as such. The api key is definitely sensitive enough, the remaining ones probably not (they don't have much value without the key) - therefore i'm leaving them as they are, i.e. plaintext.

Regarding the implementation: the BuildStartContextProcessor (updateParameters() method) is used to *add* the params, the *PasswordsProvider* (getPasswordParameters() method) only *marks* the already added params as read-only. That's how i understand the advisory i received from JetBrains (thanks you JB!)